delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViewJsonLdSourceDecoder.scala

```diff
@@ -23,15 +23,31 @@ class ElasticSearchViewJsonLdSourceDecoder private (
 
   def apply(project: Project, source: Json)(implicit
       caller: Caller
-  ): IO[ElasticSearchViewRejection, (Iri, ElasticSearchViewValue)] = decoder(project, mapJsonToString(source))
+  ): IO[ElasticSearchViewRejection, (Iri, ElasticSearchViewValue)] =
+    applyDefaultSettings(source).flatMap { s => decoder(project, mapJsonToString(s)) }
 
   def apply(project: Project, iri: Iri, source: Json)(implicit
       caller: Caller
-  ): IO[ElasticSearchViewRejection, ElasticSearchViewValue] = decoder(
-    project,
-    iri,
-    mapJsonToString(source)
-  )
+  ): IO[ElasticSearchViewRejection, ElasticSearchViewValue] =
+    applyDefaultSettings(source).flatMap { s =>
+      decoder(
+        project,
+        iri,
+        mapJsonToString(s)
+      )
+    }
+
+  /**
+    * Apply default settings if nothing has been provided
+    */
+  private def applyDefaultSettings(source: Json): UIO[Json] =
+    source.asObject match {
+      case Some(o) if !o.contains("settings") =>
```

Comment:
Working at the compacted JSON level is wrong, since this key can very well be `https://bluebrain.github.io/nexus/vocabulary/settings` and it should still work. Or it can be a curie or an alias and still work.

But I know that in few other places we have this assumption too. That's why this approach is not desirable and I suggested the other